kafka-connect-s3/src/main/java/io/confluent/connect/s3/format/avro/AvroRecordWriterProvider.java

```diff
@@ -81,6 +82,19 @@ public void write(SinkRecord record) {
           if (value instanceof NonRecordContainer) {
             value = ((NonRecordContainer) value).getValue();
           }
+          if (value == null) {
+            if (conf.nullValueBehavior().equalsIgnoreCase(BehaviorOnNullValues.IGNORE.toString())) {
+              log.debug("Null valued record cannot be written to output as Avro. "
+                  + "Skipping. Record Key: {}", record.key());
+              return;
+            } else if (conf.nullValueBehavior().equalsIgnoreCase(
+                BehaviorOnNullValues.FAIL.toString())) {
+              throw new ConnectException("Null valued records are not writeable with current "
+                  + S3SinkConnectorConfig.BEHAVIOR_ON_NULL_VALUES_CONFIG + " 'settings.");
+            } else {
+              throw new UnsupportedOperationException();
```

Comment:
That's confusing. This would be a bug that should be caught by a simple unit test. Throwing a different type of exception just for the sake of catching that bug in production doesn't seem useful to me. I suggest throwing the `ConnectException` for `FAIL`

Reply:
I'm not following -- FAIL case already throws ConnectException.

Throwing UnsupportedOperationException (or any other RuntimeException) is probably the right thing to do here. Of course testing should catch these case, but assuming on bulletproof tests is not a good strategy. This is the same precedent already set in Elasticsearch connector (they actually throw RuntimeException, but I think Unsupported is more explicit). Throwing exception in production is absolutely useful because it stops progress if you are in illegal state

Reply:
Probably I wasn't clear. My point was that `ConnectException` covers all the `else` cases, specialization doesn't seem useful, a) because this should never be executed, b) even if it did because of a bug it's more clear for the user to see a `ConnectException` rather than a `UnsupportedOperationException` that won't know what it means. Reducing boiler place might be preferable here.